### PR TITLE
[MM-20582] Migrate 'components/admin_console/team_channel_settings/ch…

### PR DESCRIPTION
**Pull request by cinlloc**
…annel' module and associated tests to TypeScript

#### Summary
Migrate 'components/admin_console/team_channel_settings/channel' module and associated tests to TypeScript.

#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/13490

### CODE REVIEW

**Review by nevyangelova — APPROVED** [review]
Thanks for the contribution @cinlloc 🎉

File: components/admin_console/team_channel_settings/channel/list/channel_list.tsx [hunk 1]
```diff
@@ -31,7 +32,7 @@ interface ChannelListProps {
     data: ChannelWithTeamData[];
     total: number;
     removeGroup?: () => void;
-    onPageChangedCallback?: () => void;
+    onPageChangedCallback?: (state: ChannelSettingsState) => void;
```
Comment (hmhealey) [inline components/admin_console/team_channel_settings/channel/list/channel_list.tsx:35]: This prop doesn't appear to actually be used, so perhaps we could remove it? But if we remove that, we're removing all the state from `ChannelSettings` so I'm not sure if I'm missing something here.

**Review by hmhealey — COMMENTED** [review]
Thanks for the PR, @cinlloc.

I'm going to defer review to @mkraft since I'm not particularly familiar with this part of the system console, but I did have a question about the below code

**Comment by jfrerich** [comment]
@mkraft, gentle reminder for review.

File: components/admin_console/team_channel_settings/channel/list/channel_list.tsx (hunk 1 above)
Comment (cinlloc) [reply components/admin_console/team_channel_settings/channel/list/channel_list.tsx]: It seems to be ok removing onPageChangedCallback from both ChannelList and ChannelSettings.

**Comment by cinlloc** [comment]
I did the fix proposed by @hmhealey in ed94e1c, could the review process continue?

**Comment by jasonblais** [comment]
Thank you @cinlloc! Would you be open to review the test failures reported in the build check? If you have any questions about them, don't hesitate to let us know :)

**Review by hmhealey — APPROVED** [review]
Sorry for the delay in the review. This definitely slipped through the cracks. Looks good to me, so it's off to QA review now

**Review by mkraft — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @cinlloc 
Tested, looks good to merge.
- Verified team channel settings and list in System Console - no change in functionality.

**Comment by hanzei** [comment]
@cinlloc Could you please merge `master` into your branch?

**Comment by hanzei** [comment]
Thank you @cinlloc :+1:

**Pull request merged**